backends/backends.h

```diff
@@ -15,6 +15,22 @@ typedef enum backend_options {
     BACKEND_OPTION_SEND_NAMES        = (1 << 16)
 } BACKEND_OPTIONS;
 
+typedef enum backend_types {
+    BACKEND_TYPE_UNKNOWN = -1,
+    BACKEND_TYPE_GRAPHITE = 0,
+    BACKEND_TYPE_OPENTSDB_TELNET = 1,
+    BACKEND_TYPE_OPENTSDB_HTTP = 2,
+    BACKEND_TYPE_JSON = 3,
+    BACKEND_TYPE_PROMETEUS = 4,
+    BACKEND_TYPE_KINESIS = 5
```

Comment:
Actual values are not used anymore, i suggest to remove -1, 0 stuff.

And one more thing about it, i think you need to add protocl suffix to all types, otherwise it is not clear why ony `opentsdb` has `http/telnet`

Reply:
All right I already changed here, so I am removing the values and allowing the compiler to give the value to them.

About the protocol suffix, I confess I did not study all the other backends, so I will rename opentsdb from:

BACKEND_TYPE_OPENTSDB_TELNET,
BACKEND_TYPE_OPENTSDB_HTTP,

to

BACKEND_TYPE_OPENTSDB_USING_TELNET,
BACKEND_TYPE_OPENTSDB_USING_HTTP,

I think this will help to let the things more clear.

Reply:
I think better to add comments in this particular case to make it more clear. But i like the direction in general - to make something clear :smile: 

so 
> TYPE_OPENTSDB_HTTP

lgtm, it is clear from the context what is `HTTP` there.

Reply:
ok, you added `USING` thing, the problem is not resolved - it is completely unclear why only types have `USING_PROTOCOL`

Reply:
All right, I will comment all the options to have the things more clear.
I already commit everything we talked, except this! :)